### ISSUE TITLE
ARIA accessibility for Select component

### ISSUE THREAD

**Issue by pavish** [centerofci/mathesar#374]
## Problem
We have a custom Select menu (`/mathesar_ui/src/components/select`), which does not have ARIA accessibility. 

## Proposed solution
We need to include ARIA accessibility to the Select menu.

w3 spec:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html
https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox

Mozilla docs:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role

Popular component with good ARIA controls for reference:
https://select2.org/

Some references:
https://adamsilver.io/blog/building-an-accessible-autocomplete-control/
https://www.24a11y.com/2019/select-your-poison/

**Comment by mr-gabe49** [comment]
I can start working on this one

**Comment by kgodey** [comment]
sounds good, @mr-gabe49!

**Comment by pavish** [comment]
@mr-gabe49 Additional svelte select components for reference:

https://github.com/rob-balfre/svelte-select
https://github.com/pavish/select-madu

**Comment by pavish** [comment]
PR #460 handles the ARIA requirements for screenreaders.

Creating new issue for handling keyboard actions mentioned in w3 spec #483

**Comment by pavish** [comment]
Reopening, since the PR #460  has [linting issues](https://github.com/centerofci/mathesar/runs/3155193872), which were visible only after merging into master.
